### ISSUE TITLE
Support parsing CREATE INDEX .. ON .. USING btree (..)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7749]
PostgreSQL has a `USING method` clause in the `CREATE INDEX` statement. The `btree` method is the default, and as such should be parsed by jOOQ if present:

----
See:
- #7745
- https://www.postgresql.org/docs/current/static/sql-createindex.html

**Comment by lukaseder** [comment]
This is implemented incorrectly. The currently implemented syntax is:

```sql
create index i on t using btree (c);
```

That's the wrong position for the clause. The correct position is:

```sql
create index i using btree on t (c);
create index i on t (c) using btree;
```

**Comment by lukaseder** [comment]
This will be fixed in #9132
